### PR TITLE
fix resource organization and dependencies

### PR DESCRIPTION
**Pull request by bridiver**
fix https://github.com/brave/brave-browser/issues/1808

brave browser implementation: https://github.com/brave/brave-browser/pull/1817

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: patches/chrome-BUILD.gn.patch [hunk 1]
```diff
@@ -123,7 +123,15 @@ index 4865a69fe984b428c4b6591d11d778e6c2669d2d..49e92b60ff1cafb14cccf620af09cd51
      ]
      output = "$target_gen_dir/chrome_exe_version.rc"
    }
-@@ -1775,6 +1799,8 @@ if (is_android) {
+@@ -1557,6 +1581,7 @@ group("resources") {
+     "//chrome/browser:resources",
+     "//chrome/common:resources",
+     "//chrome/renderer:resources",
++    "//brave:resources",
```
Comment (bridiver) [inline patches/chrome-BUILD.gn.patch]: there are chromium_src overrides that use generated header files from the resources

**Comment by NejcZdovc** [comment]
@bridiver please fix linter

**Comment by bridiver** [comment]
@NejcZdovc not sure what you're referring to

**Comment by petemill** [comment]
@bridiver We need a corresponding change in brave-browser for the rewards extension messages.json path https://github.com/brave/brave-browser/blob/master/lib/l10nUtil.js#L58

**Comment by petemill** [comment]
While we're here, 'brave_weards/resources/ui' directory feels a little weird to me, since it's only 1 part of the ui. Yes it's the thing that renders at `brave://rewards` so I guess it could be considered the 'main' UI. But is there a clearer directory name, e.g. 

```
components/brave_rewards/resources:
- extension <-- if this becomes a WebUI, it could be called 'rewards-dialog-webui'
- donate-dialog-webui
- tip-dialog-webui <-- soon
- rewards-webui <-- was 'ui'
```

**Comment by bridiver** [comment]
@petemill yep, it's weird, but I didn't want to make even more changes in one PR

**Comment by bridiver** [comment]
mainly I just wanted to get everything grouped under `resources` and in a grd file for the component

**Comment by bridiver** [comment]
I think braveComponentsResourcesPath also needs to change to use the string grd file?

**Comment by bridiver** [comment]
@bbondy anything else you see here for brave-browser changes?

**Comment by bridiver** [comment]
https://github.com/brave/brave-browser/pull/1817

**Comment by bridiver** [comment]
maybe the next change I'll make is to move that stuff from brave-browser to brave-core. These linked PRs across the repos really shouldn't be necessary

**Comment by bridiver** [comment]
I'd really like to expedite this if everything looks good to unblock @SergeyZhukovsky

**Comment by bridiver** [comment]
https://github.com/brave/brave-core/commit/331793650943b01646d3001f0c5b95de2c492f53

**Pull request merged**